pkg/statser/statser_internal.go

```diff
@@ -0,0 +1,100 @@
+package statser
+
+import (
+	"context"
+	"time"
+
+	"github.com/atlassian/gostatsd"
+)
+
+// InternalHandler is an interface to dispatch metrics to.  Exists
+// to break circular dependencies.
+type InternalHandler interface {
+	DispatchMetric(ctx context.Context, m *gostatsd.Metric) error
+}
+
+// InternalStatser is a Statser which sends metrics to a handler
+type InternalStatser struct {
+	ctx       context.Context
```

Comment:
I still think there's a better way of managing it all (the ctx and the handling of the channels being closed).  I couldn't unravel it all in my head though.